### PR TITLE
Feature/buff window

### PR DESCRIPTION
**Pull request by EvgenijK**
It would be nice to have indicators for some timed effects on character. Especially for resistances.
There is even a 4-year old topic on forum about it https://tomenet.eu/phpBB3/viewtopic.php?f=7&t=932

This code adds indicators for timed four basic resistances, poison resistance, mana resistance and timed full ESP.
Indicators placed under location coordinates, left from the map on the main window.
Because of placement they are only visible if big_map option is turned on.

image

Please leave comments if this code needs corrections to be accepted.

### CODE REVIEW

**Comment by jezek** [comment]
I see, there is new code added to server, which sends new packets when drawing to client. Does the new server code works (have you tested it) with previous clients?

**Comment by EvgenijK** [comment]
> I see, there is new code added to server, which sends new packets when drawing to client. Does the new server code works (have you tested it) with previous clients?

Thanks for pointing on this. I tested and it didn't work.

I updated game version and added client version check in server functions.

**Comment by mhirki** [comment]
I think adding 7 new packet types is a total waste. Everything should be compressed into a single new packet with 32 bits reserved for flags.

**Comment by EvgenijK** [comment]
> I think adding 7 new packet types is a total waste. Everything should be compressed into a single new packet with 32 bits reserved for flags.

I see two variants of this:
1) Send flags to toggle the indicators, i.e. if the indicator flag is in the packet, then toggle it on interface. This way requires to store flags on client.
2)  Send flags as state of indicators - if flag is set in the packet, then draw indicator, if flag isn't set, then erase it. There is no need to store info in this variant, but we have to send info about all indicators each time.

I hope i explained this variants clearly enough.
Personally I more like second variant.

What do you think, what would be better solution?

**Comment by jezek** [comment]
@EvgenijK 
> What do you think, what would be better solution?

I would opt for the second variant too.

**Comment by mhirki** [comment]
Yeah, variant 2 is defintely the answer.

**Comment by EvgenijK** [comment]
@jezek @mhirki 
I made changes, waiting for the review.

File: src/server/xtra2.c
Comment (jezek) [inline src/server/xtra2.c:620]: Indentation mismatch?

File: src/server/xtra2.c
Comment (jezek) [inline src/server/xtra2.c:2779]: Indentation mismatch?

File: src/server/xtra2.c
Comment (jezek) [inline src/server/xtra2.c:2828]: Indentation mismatch?

File: src/server/xtra2.c
Comment (jezek) [inline src/server/xtra2.c:2877]: Indentation mismatch?

File: src/server/xtra2.c
Comment (jezek) [inline src/server/xtra2.c:2926]: Indentation mismatch?

File: src/server/xtra2.c
Comment (jezek) [inline src/server/xtra2.c:2975]: Indentation mismatch?

File: src/server/xtra2.c
Comment (jezek) [inline src/server/xtra2.c:3644]: Indentation mismatch?

File: src/server/xtra1.c
Comment (jezek) [inline src/server/xtra1.c:7370]: Indentation mismatch?

File: src/client/c-xtra1.c
Comment (jezek) [inline src/client/c-xtra1.c:1031]: Indentation mismatch?

File: src/client/nclient.c
Comment (jezek) [inline src/client/nclient.c:181]: Indentation mismatch?

File: src/client/nclient.c
Comment (jezek) [inline src/client/nclient.c:4846]: Indentation mismatch?

File: src/common/defines.h
Comment (jezek) [inline src/common/defines.h:2227]: Indentation consistency?

File: src/server/nserver.c
Comment (jezek) [inline src/server/nserver.c:8854]: Indentation mismatch?

File: src/server/xtra1.c
Comment (jezek) [inline src/server/xtra1.c:688]: Indentation mismatch?

File: src/common/pack.h
Comment (jezek) [inline src/common/pack.h:256]: I know, there are many misaligned constants, but try to align as the one before.

**Review by jezek — CHANGES_REQUESTED** [review]
Apart from some indentation consistency mismatch, I see no logical flaws.
Note: I haven't tested if it compiles or behaves as expected (yet).

File: src/server/xtra2.c
Comment (EvgenijK) [reply src/server/xtra2.c]: fixed

File: src/server/xtra2.c
Comment (EvgenijK) [reply src/server/xtra2.c]: fixed

File: src/server/xtra2.c
Comment (EvgenijK) [reply src/server/xtra2.c]: fixed

File: src/server/xtra2.c
Comment (EvgenijK) [reply src/server/xtra2.c]: fixed

File: src/server/xtra2.c
Comment (EvgenijK) [reply src/server/xtra2.c]: fixed

File: src/client/nclient.c
Comment (EvgenijK) [reply src/client/nclient.c]: fixed

File: src/server/nserver.c
Comment (EvgenijK) [reply src/server/nserver.c]: fixed

File: src/server/xtra1.c
Comment (EvgenijK) [reply src/server/xtra1.c]: fixed

File: src/common/pack.h
Comment (EvgenijK) [reply src/common/pack.h]: fixed

File: src/common/defines.h
Comment (EvgenijK) [reply src/common/defines.h:2227]: Tried to fixe this in CLion, Vim and github online editor, all of them shows that `25` on the same columt with other values

Comment (jezek) [reply src/common/defines.h:2227]: That's wierd, but you're right. When I look at the [whole file](https://github.com/TomenetGame/tomenet/blob/e8fcb013e0297dad3f39141cddeb686559e299f3/src/common/defines.h#L2224-L2227), the indetation is ok.

File: src/server/nserver.c
Comment (jezek) [inline src/server/nserver.c:8860]: I would make this a oneliner. Other lines above are longer, so it wouldn't stand out and wrapping can do an editor.

**Review by jezek — CHANGES_REQUESTED** [review]
Thank you for indentation change. Only the 2 line logging caught my eye. Sorry for missing it the first time and is just a suggestion of me. The maintainers have the last word, so let's wait for their comments. When I have more free time, I will try to compile (server, client) and test it,

**Comment by jezek** [comment]
I've tested this PR. After merging to current mater, it compiles (server+client) on ubuntu 20.10 without introducing new warnings. I've also tested new (4.7.3.1) server against new (4.7.3.1) and old (4.7.3) client and vice versa. Everything works as described in this PR, no bugs spotted.

**Comment by mhirki** [comment]
Thanks for the testing jezek. I did my own testing and everything seems to work. I'm gonna merge the code now.

**Pull request merged**
